website/content/en/docs/cloud-providers/AWS/launch-templates.md

```diff
@@ -0,0 +1,236 @@
+
+Certain fields are obviously critical, such as AMI and User Data. Some fields are useful for particular workloads, such as storage and IAM Instance Profile. 
+
+Finally, **the majority of Launch Template fields should not be set** (or will have no effect), such as network interfaces and instance type. 
+
+## Important Fields
+
+When creating a custom launch template, the AMI and User Data are the defining characteristics. 
+
+### AMI
+
+AMI (Amazon Machine Image), is the base image/VM for a launch template.
+
+[Review the instructions for importing a VM to AWS.](https://docs.aws.amazon.com/vm-import/latest/userguide/vmimport-image-import.html) Note the AMI id generated by this process, such as, 
+`ami-074cce78125f09d61`.
+
+### User Data - Autoconfigure
+
+Importantly, the AMI must support automatically connecting to a cluster based
+on "user data", or a base64 encoded string passed to the instance at startup.
+The syntax and purpose of the user data varies between images. The Karpenter
+default OS, Amazon Linux 2 (AL2), accepts shell scripts (bash commands). 
+
+[AWS calls data passed to an instance at launch time "user
+data".](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html#user-data-shell-scripts)
+
+In the default configuration, Karpenter uses an EKS optimized version of AL2 and passes the hostname of the Kubernetes API server, and a certificate. The EKS Optimized AMI includes a `bootstrap.sh` script which connects the instance to the cluster, based on the passed data. 
+
+Alternatively, you may reference AWS's [`bootstrap.sh`
+file](https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh)
+when building a custom base image. 
+
+```
+#!/bin/bash
+/etc/eks/bootstrap.sh <my-cluster-name> \
+--kubelet-extra-args <'--max-pods=40'> \
+--b64-cluster-ca <certificateAuthority> \
+--apiserver-endpoint <endpoint> 
+--dns-cluster-ip <serviceIpv4Cidr>
+--use-max-pods false
+```
+
+Note, you must populate this command with live values. Karpenter will
+not change the user data in the launch template. 
+
+Encode using yaml function `!Base64` yaml function or `cat userdata.sh | base64 > userdata-encoded.txt` shell command. 
+
+**Bootstrap Script Parameters**
+
+The sample bootstrap script requires information to join the cluster.
+
+These values may be found using:
+```
+aws eks describe-cluster --name MyKarpenterCluster 
+```
+
+**Kubelet Arguments**
+
+Specifying max-pods can break Karpenter's binpacking logic (it has no way to know what this setting is). If karpenter attempts to pack more than this number of pods, the instance may be oversized, and additional pods will reschedule.
+
+## Situational Fields 
+
+Configure these values in response to a particular use case, such as nodes interacting with another AWS service, or using EBS storage on the node. 
+
+### Instance Profile - IAM
+
+The launch template must include an "instance profile" -- a set of IAM roles. 

```

Comment:
`a set of IAM policies` instance profiles are 1:1 w/ roles.

Reply:
<img width="529" alt="Screen Shot 2021-11-09 at 9 14 36 AM" src="https://user-images.githubusercontent.com/5782953/140950977-d55bbdb8-a6f7-43df-832d-909cfefd6af9.png">

To me, this looks like "list of roles" but perhaps I'm interpreting it wrong?

Reply:
Very confusing, but InstanceProfile's can only have 1 role:

https://awscli.amazonaws.com/v2/documentation/api/latest/reference/iam/add-role-to-instance-profile.html#description

```
Adds the specified IAM role to the specified instance profile. An instance profile can contain only one role, and this quota cannot be increased
```

Reply:
*screams*